pennylane/ops/op_math/sum.py

```diff
@@ -0,0 +1,286 @@
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains the implementation of the Sum class which contains logic for
+computing the sum of operations.
+"""
+from functools import reduce
+
+import numpy as np
+import pennylane as qml
+
+from pennylane import math
+from pennylane.operation import Operator
+
+
+def op_sum(*summands, do_queue=True, id=None):
+    r"""Construct an operator which is the sum of the given operators.
+
+    Args:
+        *summands (tuple[~.operation.Operator]): the operators we want to sum together.
+
+    Keyword Args:
+        do_queue (bool): determines if the sum operator will be queued (currently not supported).
+            Default is True.
+        id (str or None): id for the sum operator. Default is None.
+
+    Returns:
+        ~ops.op_math.Sum: the operator representing the sum of summands.
+
+    ..seealso:: :class:`~.ops.op_math.Sum`
+
+    **Example**
+
+    >>> summed_op = op_sum(qml.PauliX(0), qml.PauliZ(0))
+    >>> summed_op
+    PauliX(wires=[0]) + PauliZ(wires=[0])
+    >>> summed_op.matrix()
+    array([[ 1,  1],
+           [ 1, -1]])
+    """
+    return Sum(*summands, do_queue=do_queue, id=id)
+
+
+def _sum(mats_gen, dtype=None, cast_like=None):
+    r"""Private method to compute the sum of matrices.
+
+    Args:
+        mats_gen (Generator): a python generator which produces the matricies which
+            will be summed together.
+
+    Keyword Args:
+        dtype (str): a string representing the data type of the entries in the result.
+        cast_like (Tensor): a tensor with the desired data type in its entries.
+
+    Returns:
+        res (Tensor): the tensor which is the sum of the matrices obtained from mats_gen.
+    """
+    # Note this method is currently inefficient (improve addition by looking at wire subgroups)
+    res = reduce(math.add, mats_gen)
+
+    if dtype is not None:
+        res = math.cast(res, dtype)
+    if cast_like is not None:
+        res = math.cast_like(res, cast_like)
+
+    return res
+
+
+class Sum(Operator):
+    r"""Symbolic operator representing the sum of operators.
+
+    Args:
+        summands (tuple[~.operation.Operators]): a tuple of operators which will be summed together.
+
+    Keyword Args:
+        do_queue (bool): determines if the sum operator will be queued (currently not supported).
+            Default is True.
+        id (str or None): id for the sum operator. Default is None.
+
+    Returns:
+        ~ops.op_math.Sum: the operator representing the sum of summands.
+
+    ..seealso:: :class:`~.ops.op_math.Sum`
+
+    **Example**
+
+    >>> summed_op = op_sum(qml.PauliX(0), qml.PauliZ(0))
```

Comment:
```suggestion
    >>> summed_op = Sum(qml.PauliX(0), qml.PauliZ(0))
```

I think it's fine to directly construct the class in the docstring for the class.